src/components/CompositeHeader/CompositeHeader.jsx

```diff
@@ -7,12 +7,26 @@ import Icon, { iconPropType } from 'cozy-ui/transpiled/react/Icon'
 
 // TODO Test and improve it & PR cozy-ui
 const CompositeHeader = ({ icon, title, text, className, ...restProps }) => {
+  const [imgScr, setImgSrc] = React.useState('')
+
+  React.useEffect(() => {
```

Comment:
We can do just that ?
```
useEffect(() => {
const loadImg = async () => {
  const src = await import(`src/assets/icons/${icon}`)
  setImgSrc(src)
}
loadImg()
}, [icon])
```

Reply:
Pour info c'est un des premiers hooks que j'ai prévu d'implémenter dans la lib cozy-hooks (cf tech transverse)

Plus d'info: https://github.com/cozy/cozy-libs/pull/1340#discussion_r692691754

Reply:
Je comprends pas le pb de fuite mémoire que ça peut engendrer. Tu peux expliquer stp ? :)

Reply:
> Pour info c'est un des premiers hooks que j'ai prévu d'implémenter dans la lib cozy-hooks (cf tech transverse)

J'ai hâte 😁 

>Je comprends pas le pb de fuite mémoire que ça peut engendrer. Tu peux expliquer stp ? :)

Si ton composant est démonté alors que ton appel asynchrone est lancé mais pas encore résolu, une fois ce dernier résolu React va tenter de mettre à jour un state qui n'existe plus 😞 
Et tu auras ce genre de warning:
![Capture d’écran 2021-08-31 à 16 31 37](https://user-images.githubusercontent.com/14182143/131521743-f791caca-7fd4-4c73-8f45-fa8069ff2b78.png)

Ce "fix" permet d'éviter que cela arrive 👍

Reply:
Ah d'accord, merci pour l'explication.